### PR TITLE
Allow to enable/disable guests and other extra config settings

### PR DESCRIPTION
**Pull request by jespino**
#### Summary
Allow to enable/disable guests and other extra config settings

### CODE REVIEW

**Comment by grundleborg** [comment]
@jespino this PR has a huge diff - is there a bad merge here or is it meant to have everything rolled into it?

**Comment by jespino** [comment]
@grundleborg Yes, I screwed this thing merging master instead the correct branch. I'll fix it.

**Comment by jespino** [comment]
@grundleborg fixed :)

File: api4/user.go [hunk 1]
```diff
@@ -94,7 +94,26 @@ func createUser(c *Context, w http.ResponseWriter, r *http.Request) {
...
-		ruser, err = c.App.CreateUserWithToken(user, tokenId)
+		var token *model.Token
+		token, err = c.App.Srv.Store.Token().GetByToken(tokenId)
+		if err != nil {
+			c.Err = model.NewAppError("CreateUserWithToken", "api.user.create_user.signup_link_invalid.app_error", nil, err.Error(), http.StatusBadRequest)
+			return
+		}
+
+		if token.Type == app.TOKEN_TYPE_GUEST_INVITATION {
+			if c.App.License() == nil {
```
Comment (grundleborg) [inline api4/user.go]: Should there also be a check for a licence feature here?

Comment (jespino) [reply api4/user.go]: It is not an specific feature flag for that, have a license means that you are able to use guest accounts, but I think you are right, we should be checking another feature flag there. I going to create another ticket to handle it in all the code.

File: api4/user.go [hunk 2]
```diff
@@ -94,7 +94,26 @@ func createUser(c *Context, w http.ResponseWriter, r *http.Request) {
...
+				return
+			}
+			if !*c.App.Config().GuestAccountsSettings.Enable {
+				c.Err = model.NewAppError("CreateUserWithToken", "api.user.create_user.guest_accounts.disabled.app_error", nil, "", http.StatusBadRequest)
+				return
+			}
+		}
+		ruser, err = c.App.CreateUserWithToken(user, token)
+		if err == nil {
+		}
```
Comment (grundleborg) [inline api4/user.go]: What's this?

Comment (jespino) [reply api4/user.go]: A typo :man_facepalming:

Comment (jespino) [reply api4/user.go]: Fixed

**Pull request merged**